db/sqlglue.c

```diff
@@ -1717,8 +1742,10 @@ int create_datacopy_array(struct dbtable *tbl)
             return -1;
         }
 
-        if (!(schema->flags & SCHEMA_DATACOPY)) {
+        if (!(schema->flags & (SCHEMA_DATACOPY | SCHEMA_PARTIALDATACOPY))) {
             continue;
+        } else if (schema->flags & SCHEMA_PARTIALDATACOPY) {
+            ondisk = schema->partial_datacopy;
```

Comment:
In create_datacopy_array I set the ondisk to only include the partial datacopy columns rather than all of the columns. Would this be sufficient or are there areas that I missed where the full data is still being saved?

Reply:
@riverszhang89

Reply:
I don't think it's sufficient. I think you'd handle this in `db/record.c` (e.g., `add_record()`, `upd_record()`, etc.) - instead of writing the entire ondisk data alongside with the index entry, only the partially datacopied fields would be saved.

Reply:
Ah ok I'll look into that, thanks!